src/messages/MOSDMap.h

```diff
@@ -158,6 +158,10 @@ class MOSDMap final : public Message {
       out << " src has " << oldest_map << ".." << newest_map;
     out << ")";
   }
+
+  bool support_reencode() const override {
```

Comment:
i don't think `suppoert_reencode()` is not the right word. per my understanding, `MOSDMap` actually can be reencoded. what prevents it from joining the club of messages which can be fast prepared is just that encoding a MOSDMap is CPU bound. so we would prefer do this in messenger's write thread instead of doing so using the fast preparing machinery. 

so i'd suggest change the name to `can_fast_encode()`.